docs/proposals/permission-level-degradation.md

```diff
@@ -0,0 +1,188 @@
+
+When users define their own `Roles` within their Project or Organization, bulward should not update those `Roles` to "fix" roles that violate the max permission level.
+Changing user defined specification is very confusing to administrators and hard to make transparent to users.
+
+Instead we should define our own Role/RoleBinding objects for each permission scope (Project/Organization) and translate that into Kubernetes `Role`/`RoleBinding` objects.
+This way we can keep the user specification intact, while enabling only parts of his specification when it violates the max permission level.
+We can also report detailed 
+
+### New CRDs
+
+## ProjectRole
+
+`ProjectRole` objects control a `Role` object within the `Project` namespace, that the object was created in.
+In difference to Kubernetes `Role` objects, Bulward will only consider RBAC rules in the definition that are not in violation of the max permission level of the project.
+
+The accepted rules will be listed in the objects status to inform the user.
+
+`ProjectRoles` are validated on creation and update to enforce the same prerequisites as Kubernetes:
+https://kubernetes.io/docs/reference/access-authn-authz/rbac/#restrictions-on-role-creation-or-update
+
+```yaml
+apiVersion: bulward.io/v1alpha1
+kind: ProjectRole
+metadata:
+  name: mycoolapps-admin
+  namespace: project-namespace
+rules:
+- apiGroups:
+  - my-corp.com
+  resources:
+  - mycoolapps
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
+status:
+  phase: Established
+  observedGeneration: 1
+  conditions: []
+  acceptedRules:
+  - apiGroups:
+    - my-corp.com
+    resources:
+    - mycoolapps
+    verbs: # limited verbs, due to a max permission issue
+    - get
+    - list
+    - watch
+    - update
+    - patch
+```
+
+## OrganizationRole
+
+`OrganizationRole` objects control a `Role` object within the `Organization` namespace and all `Project` namespaces existing within the organization.
+In difference to Kubernetes `Role` objects, Bulward will only consider RBAC rules in the definition that are not in violation of the max permission level of the organization.
+
+The accepted rules will be listed in the objects status to inform the user.
+
+`OrganizationRoles` are validated on creation and update to enforce the same prerequisites as Kubernetes:
+https://kubernetes.io/docs/reference/access-authn-authz/rbac/#restrictions-on-role-creation-or-update
+
+```yaml
+apiVersion: bulward.io/v1alpha1
+kind: OrganizationRole
+metadata:
+  name: mycoolapps-admin
+rules:
+- apiGroups:
+  - my-corp.com
+  resources:
+  - mycoolapps
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
+status:
+  phase: Established
+  observedGeneration: 1
+  conditions: []
+  acceptedRules:
+  - apiGroups:
+    - my-corp.com
+    resources:
+    - mycoolapps
+    verbs:
+    - get
+    - list
+    - watch
+    - update
+    - patch
+```
+
+## ProjectRoleBinding
```

Comment:
Why is this needed? Why ordinary rolebinding aren't enough?

Reply:
Because we want custom validation, at least to make sure that `ProjectRoleBinding` objects can only reference `ProjectRoles` and `OrganizationRoles` and `OrganizationRoleBindings` can only reference `OrganizationRoles`.

We don't want someone to try to bind directly to `Roles`/`ClusterRoles`